src/consensus/pbft/libbyz/replica.cpp

```diff
@@ -518,14 +569,19 @@ void Replica::playback_pre_prepare(ccf::Store::Tx& tx)
 
     last_te_version = ledger_writer->write_pre_prepare(tx, executable_pp.get());
 
+    global_commit(executable_pp.get());
```

Comment:
why does this need to occur before last_executed++?

Reply:
because we open the network on global commit, and we do `seqno_at_last_f_change = last_executed + 1` there, so we should not update `last_executed` before that